eZ/Publish/Core/Search/Common/IterativelyIndexer.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Search\Common;
+
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Helper\ProgressBar;
+use PDO;
+
+/**
+ * Base class for the Search Engine Indexer Service aimed to recreate Search Engine Index.
+ * Each Search Engine has to extend it on its own.
+ *
+ * Extends indexer to allow for reindexing your install while it is in production by splitting indexing into tree tasks:
+ * - Remove items in index no longer valid in database
+ * - Making purge of index optional
+ * - indexing by specifying id's, for purpose of supporting parallel indexing
+ *
+ * @api
+ */
+abstract class IterativelyIndexer extends Indexer
```

Comment:
Name looks strange to me. Rather `Iterative...` but if the point of this, as I understand, is to index new things keeping old ones, then maybe `IncrementalIndexer` (based on "incremental backup" present in e.g. SQL Server :grinning: )

Reply:
I like the name